lib/actions/user.js

```diff
@@ -28,102 +24,135 @@ function createNewUser (auth0User) {
 }
 
 /**
- * Fetches the saved/monitored trips for a user.
- * We use the accessToken to fetch the data regardless of
- * whether the process to populate state.user is completed or not.
+ * Helper function that
+ * - extracts key variables from the state and passes them to the code to execute:
+ *     - apiKey and apiBaseUrl from state.otp.config.persistence.otp_middleware,
+ *     - accessToken and loggedIUnUser from state.user.
+ * - checks that otp_middleware is set, and throws an error if not.
+ * @param functionToExecute the code to execute, with parameters (dispatch, arguments)
```

Comment:
Please note that `functionToExecute` can be a function that can be awaited upon.